src/systemctl/systemctl-start-special.c

```diff
@@ -197,24 +197,35 @@ int verb_start_special(int argc, char *argv[], void *userdata) {
                            ACTION_POWEROFF,
                            ACTION_REBOOT,
                            ACTION_KEXEC,
-                           ACTION_HALT,
-                           ACTION_SUSPEND,
-                           ACTION_HIBERNATE,
-                           ACTION_HYBRID_SLEEP,
-                           ACTION_SUSPEND_THEN_HIBERNATE)) {
+                           ACTION_HALT)) {
+
+                        if (arg_when == 0)
+                                r = logind_cancel_shutdown();
+                        else if (arg_when != USEC_INFINITY)
+                                r = logind_schedule_shutdown(a);
+                        else
+                                r = logind_reboot(a);
```

Comment:
Hmm, canceling action with `0` timespan sounds spurious for me...
Maybe, better to swap the conditions:
```c
if (arg_when == 0)
        r = logind_reboot(a);
else if (arg_when != USEC_INFINITY)
        r = logind_schedule_shutdown(a);
else /* arg_when == USEC_INFINITY */
        r = logind_cancel_shutdown();
```